app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,31 @@
+class CommentsController < ApplicationController
+  before_action :set_event
+  before_action :auth_user?
+
+
+  def create
+    @comment = @event.comments.new(comment_params)
+    @comment.user = current_user
+    @comment.save
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy
+    @comment = @event.comments.find(params[:id])
+    @comment.destroy
+    redirect_back(fallback_location: root_path)
+  end
+  
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:body, :event_id)
+  end
+
+  def set_event
+    @event = Event.find(params[:event_id])
```

Comment:
Add a guard if the event is null

Reply:
For security, I already put this line of code on each model comment and answer : "validates :body, presence: true, allow_blank: false" so the comment will not be created if it is empty.
For the null, an answer depends on a comment and a comment depends on an event, so if the event is null, a comment can't exist, no way to create one. and with the event, in the create method, there is already a condition.

Reply:
got it